include/no-os/irq.h

```diff
@@ -85,6 +92,8 @@ struct irq_platform_ops ;
 struct irq_init_param {
 	/** Interrupt request controller ID. */
 	uint32_t	irq_ctrl_id;
+	/** Interrupt source peripheral. */
+	enum irq_source source;
```

Comment:
As stated in the conversation section, I believe this shouldn't feature in the interrupt controller initialization, because we would need one interrupt controller to initialize once and service all interrupts it has.

Reply:
true, this should probably go in the callback descriptor then